File: dsa-interview-crash-course/1413_min_value_to_get_positiv_step_by_step_sum.cpp
#include <algorithm>
#include <numeric>
#include <vector>

class Solution {
   public:
    int minStartValue(std::vector<int>& nums) {
        std::vector<int> cum_sum(nums.size(), 0);
        std::partial_sum(nums.begin(), nums.end(), cum_sum.begin());
        int min_val = *std::min_element(cum_sum.begin(), cum_sum.end());
        return min_val > 0 ? 1 : abs(min_val) + 1;
    }
};

File: dsa-interview-crash-course/1413_min_value_to_get_positiv_step_by_step_sum.py
from itertools import accumulate
from typing import List


class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        cum_sum = list(accumulate(nums))
        min_val = min(cum_sum)
        return 1 if min_val > 0 else (abs(min_val) + 1) 